### ISSUE TITLE
New cop to detect unnecessary hash merge and then splat

### ISSUE THREAD

**Issue by fatkodima** [rubocop/rubocop#12257]
Just found an example code from rails that can be improved https://github.com/rails/rails/blob/8ab5e58272b1da9548bb607542525002b44799ed/activesupport/lib/active_support/number_helper/number_converter.rb#L166-L172

Duplicating first method here
```ruby
# Bad
def translate_number_value_with_default(key, **i18n_options)
  I18n.translate(key, **{ default: default_value(key), scope: :number }.merge!(i18n_options))
end
```

There is unnecessary `merge!` and hash splat. Can be also written as:
```ruby
# Good
def translate_number_value_with_default(key, **i18n_options)
  I18n.translate(key, default: default_value(key), scope: :number, **i18n_options)
end
```

I have also seen such examples before in the wild, so this is quite a popular pattern.
